trie/stacktrie.go

```diff
@@ -345,68 +360,44 @@ func (t *StackTrie) hash(st *stNode, path []byte) {
 		n := shortNode{Key: hexToCompactInPlace(st.key), Val: valueNode(st.val)}
 
 		n.encode(t.h.encbuf)
-		encodedNode = t.h.encodedBytes()
+		blob = t.h.encodedBytes()
 
 	default:
 		panic("invalid node type")
 	}
 
 	st.typ = hashedNode
 	st.key = st.key[:0]
-	if len(encodedNode) < 32 {
-		st.val = common.CopyBytes(encodedNode)
+
+	// Skip committing if the non-root node if the size is smaller than 32 bytes.
+	if len(blob) < 32 && len(prefix) > 0 {
+		st.val = common.CopyBytes(blob)
```

Comment:
Ah, so here you modify the clause to hash the thing if it is root (len prefix > 0), so thus you can skip the force-hash.  I see what you did there.. 
Yes this probably works nicely.